perfkitbenchmarker/linux_benchmarks/memcached_ycsb_benchmark.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs YCSB against memcached.
+
+This benchmark runs two workloads against memcached using YCSB (the Yahoo! Cloud
+Serving Benchmark).
+memcached is described in perfkitbenchmarker.linux_packages.memcached_server
+YCSB and workloads described in perfkitbenchmarker.linux_packages.ycsb.
+"""
+
+import functools
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import memcached_server
+from perfkitbenchmarker.linux_packages import ycsb
+
+FLAGS = flags.FLAGS
+
+
+BENCHMARK_NAME = 'memcached_ycsb'
+BENCHMARK_CONFIG = """
+memcached_ycsb:
+  description: >
+    Run YCSB against an memcached
+    installation. Specify the number of YCSB VMs with
+    --ycsb_client_vms.
+  vm_groups:
+    servers:
+      vm_spec: *default_single_core
+      vm_count: null
+    clients:
+      vm_spec: *default_single_core
+"""
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  ycsb.CheckPrerequisites()
+
+
+def Prepare(benchmark_spec):
+  """Prepare the virtual machines to run YCSB against memcached.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  loaders = benchmark_spec.vm_groups['clients']
+  assert loaders, benchmark_spec.vm_groups
+
+  # Memcached cluster
+  memcached_vms = benchmark_spec.vm_groups['servers']
+  assert memcached_vms, 'No memcached VMs: {0}'.format(
+      benchmark_spec.vm_groups)
+
+  memcached_install_fns = [functools.partial(memcached_server.ConfigureAndStart,
+                                             vm)
+                           for vm in memcached_vms]
+  ycsb_install_fns = [functools.partial(vm.Install, 'ycsb')
+                      for vm in loaders]
+
+  vm_util.RunThreaded(lambda f: f(), memcached_install_fns + ycsb_install_fns)
+  benchmark_spec.executor = ycsb.YCSBExecutor(
+      'memcached',
+      **{'memcached.hosts': memcached_vms[0].internal_ip})
```

Comment:
How many memcached servers?
From the code, seems like there is no traffic between memcached servers. And this benchmark only use first memcached_vms as ycsb target.